pkg/collector/corechecks/containers/kubelet/provider/pod/provider.go

```diff
@@ -70,7 +70,7 @@ func (p *Provider) Provide(kc kubelet.KubeUtilInterface, sender sender.Sender) e
 	sender.Gauge(common.KubeletMetricsPrefix+"pods.expired", float64(pods.ExpiredCount), "", p.config.Tags)
 
 	for _, pod := range pods.Items {
-		//for _, container := range pod.Spec.Containers {
+		common.CachePodTagsByPVC(pod)
```

Comment:
I'm wondering why we need to push the pod tag in a cache instead of getting them locally in this for loop? 
do we reuse them in another part of the code?

What I see in this `CachePodTagsByPVC` function is that we always set the entry in the cache even it is already in, so I don't know if we improve performance by caching this info.

Reply:
They're inserted into the cache here, and fetched from the cache for the volume metrics transformed from the /metrics endpoint.

I opted for a cache to keep the interface simple and to reduce the need to refactor the function signatures of all of the metric providers. But I can also just as easily add this map as a field on the pod provider and the /metrics provider, and it could accomplish the same thing. Using the cache allowed me to also take advantage of the existing expiration timer logic built into that so that I did not need to re-implement something for this.

The cache here is not being used for performance reasons, it is being used to share data, which could be a good reason to not actually use a cache then over a map. As far as always setting the entry, it is for 2 reasons:
1) in case pod tags change
2) to refresh the expiration timer on the cached entry

Reply:
I have updated the PR and removed the cache, replacing it instead with a map that is shared throughout the course of the check and is cleared after every check run. Let me know what you think